MvvmCross.Android.Support/V7.AppCompat/StartupLifecycleCallback.cs

```diff
@@ -0,0 +1,262 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MS-PL license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Android.App;
+using Android.OS;
+using MvvmCross.Core;
+using MvvmCross.Platforms.Android;
+using MvvmCross.Platforms.Android.Core;
+using MvvmCross.ViewModels;
+
+namespace MvvmCross.Droid.Support.V7.AppCompat
+{
+    /// <summary>
+    /// Would be good to move this into core
+    /// </summary>
+    public enum ApplicationState
+    {
+        Unknown,
+        Deactivation,   //Active => inactive
+        Background,     //Start running in background (either from foreground or inactive state)
+        Activation,     //Inative => active
+        Foreground,     //Background => Foreground
+    }
+
+    /// <summary>
+    /// Mark all startup activities with this interface.
+    /// A startup activity is an activity that may be launched directly by Android using an intent filter.
+    /// </summary>
+    public interface IStartupActivity
+    {
+        /// <summary>
+        /// Called after the startup lifecycle has finished. 
+        /// The startup activity, whichever it is, can control what it should do in this method.
+        /// </summary>
+        /// <remarks>
+        /// Typical usage 1: 
+        /// if a statup activity is the "[MainActivity]" and is a splash screen which don't have a corresponding viewmodel,
+        /// close it by calling Finish() in this method. Otherwise leave blank.
+        /// 
+        /// Typical usage 2: 
+        /// if a startup activity is started by an intent filter and is NOT the mainactivity, it won't have a corresponding viewmodel.
+        /// close it by calling Finish() in this method. Otherwise leave blank.
+        /// 
+        /// Note that by default any activity can be started by another app using an intent filter targeting your app. If this happens and the activities are not decorated with IStartupActivity, mvvmcross won't work as expected.
+        /// </remarks>
+        void FinishActivity();
+    }
+
+    /// <summary>
+    /// Mark all your root activities with this interface.
+    /// </summary>
+    /// <remarks>
+    /// A root activity shuts down the mvvmcross application to free resources when it is destroyed.
+    /// </remarks>
+    public interface IShutdownActivity
+    {
+    }
+
+    public class StartupLifecycleCallback : Java.Lang.Object, Application.IActivityLifecycleCallbacks, IMvxSetupMonitor, IMvxAndroidCurrentTopActivity
```

Comment:
Needs `Mvx` naming convention.

Reply:
You mean prefix with Mvx ?

Reply:
Yes. This would count for any custom class.